lms/djangoapps/certificates/models.py

```diff
@@ -370,17 +372,23 @@ def _revoke_certificate(self, status, mode=None, grade=None, source=None):
         if not mode:
             mode = self.mode
 
-        profile_name = student_api.get_name(self.user.id)
-        if not profile_name:
-            profile_name = ''
+        preferred_name = student_api.get_name(self.user.id)
```

Comment:
Could we have this call the exiting _get_preferred_certificate_name()_ utility method?

Reply:
This would introduce a circular dependency unfortunately - I would have to move the method somewhere else (maybe a separate "utils" file, I'm not sure what would make the most sense)

Reply:
One of those. Fun!

Could we leave this code in this file but make it its own method, maybe?

Sorry to be a pain. Ignore me if you just need to get this merged.

Reply:
You're not, that makes sense! I'll do that before I merge.